etcdserver/api/v2store/node_extern_test.go

```diff
@@ -104,5 +104,5 @@ func TestNodeExternClone(t *testing.T) {
 func sameSlice(a, b []*NodeExtern) bool {
 	ah := (*reflect.SliceHeader)(unsafe.Pointer(&a))
 	bh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
-	return *ah == *bh
+	return ah.Data == bh.Data && ah.Len == bh.Len && ah.Cap == bh.Cap
```

Comment:
cherry pick this commit to make code same as in 3.5 https://github.com/etcd-io/etcd/pull/14134/commits/300c90a5e6f0b2ecc02841510860ffbec5caf8bc

Reply:
We do not necessarily backport everything from 3.5 to 3.4. 

And the previously fix isn't Semantically correct. The goal of the function `sameSlice` is to check whether the two slices are the same or not.  

If you look at go runtime's source code, you will get that a slice is actually a struct in golang, and it contains three fields. The change what I did is exactly to check the three fields. Actually the original code (`*ah ==  *bh`) is also correct to compare a struct, but go vet complains it because it's `reflect.SliceHeader`.

The fix in 3.5 * main (see below) isn't semantically correct, because [va.Pointer() ](https://pkg.go.dev/reflect#Value.Pointer)returns the first element of the slice. It's possible that two slices are different, but the first element is the same. I will submit a PR to fix it in main and 3.5. 

```
va := reflect.ValueOf(a)
vb := reflect.ValueOf(b)
return va.Len() == vb.Len() && va.Pointer() == vb.Pointer()
```

Reply:
Correction to my previous comment. The code in 3.5 & main is also correct. The [doc](https://pkg.go.dev/reflect#Value.Pointer) mentions that `If v's Kind is Slice, the returned pointer is to the first element of the slice`. Actually it's the first element in the underneath array. The memory space for an array must be continuous, so if the first element is the same, then it means they are pointing to the same array.